### ISSUE TITLE
get empty after set unless isReady

### ISSUE THREAD

**Issue by ronag** [deepstreamIO/deepstream.io-client-js#127]
This is behaviour is not consistent with the examples...

```
      const file = {
        record: ds.record.getRecord(`file/${id}`)
      }

      // await new Promise(::file.record.whenReady)

      file.record.set({
        writing: true,
        written: 0,
        size: size || null
      })

      console.log(file.record.get())
```

Which will print `{}` unless `whenReady` is waited for...?

**Comment by yasserf** [comment]
Can you point me towards the example your talking about?

**Comment by ronag** [comment]
get(path) @ https://www.deepstream.io/docs/record.html

**Comment by ronag** [comment]
It should probably also be documented in `set` that mutations are queued until `isReady`.

**Comment by ronag** [comment]
@yasserf: What is the motivation behind this behaviour? In my mind any local set should be immediately available locally and remote updates should cause a conflict which can be handled by the merge strategy.

**Comment by yasserf** [comment]
That's a good point, since once we implement offline first functionality we would expect records to be created and set accordingly.

Merge strategies were just implemented ( currently in our 1.0 beta ), so this functionality existed prior to solve that issue. We will be addressing these issues once we focus on offline first functionality.

Regarding documentation, that's a good catch. We're actually doing a large website update in the very near future and will redoing tutorials to make them less confusing and more example orientated. If you have any doc issues please raise them against the [deepstream.io-website](https://github.com/deepstreamIO/deepstream.io-website/issues) so we can track all required efforts!
